src/Sylius/Behat/Context/Setup/ShippingContext.php

```diff
@@ -183,6 +183,11 @@ private function createShippingMethod(
 
         $this->shippingMethodRepository->add($shippingMethod);
         $this->sharedStorage->set('shipping_method', $shippingMethod);
+
+        if ($this->sharedStorage->has('channel')) {
+            $channel = $this->sharedStorage->get('channel');
+            $channel->addShippingMethod($shippingMethod);
+        }
```

Comment:
This `if` should be before `$this->shippingMethodRepository->add($shippingMethod);`. Right now the relation is not flushed in this step.